File: problems/tier2-intermediate/012-elevator-system/boilerplate/cpp/part1/guided.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;


// Data class (given).
struct ElevOp {
    string kind;
    string s1;
    int i1;
    int i2;
    ElevOp(const string& kind_, const string& s1_ = "", int i1_ = 0, int i2_ = 0)
      : kind(kind_), s1(s1_), i1(i1_), i2(i2_) {}
};

// HINT: introduce an abstraction so new variants don't change existing code.
// HINT: keep each piece small — one responsibility per class.

// HINT: start from what this must return, then work backwards to the state it needs.
vector<string> elevator_simulate(vector<ElevOp> ops) {
    // TODO: write your solution
    return {};
}

File: problems/tier2-intermediate/012-elevator-system/boilerplate/java/part1/Guided.java
import java.util.*;

// Data class (given).
class ElevOp {
    public String kind;
    public String s1;
    public int i1;
    public int i2;

    public ElevOp(String kind, String s1, int i1, int i2) {
        this.kind = kind;
        this.s1 = s1;
        this.i1 = i1;
        this.i2 = i2;
    }

    public ElevOp(String kind) {
        this(kind, "", 0, 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> elevator_simulate(List<ElevOp> ops) {
        // TODO: write your solution
        return null;
    }

}
